06.test-ismb-data.py

```diff
@@ -0,0 +1,52 @@
+import sys
+import utils
+import pickle
+import argparse
+import requests
+import numpy as np
+import pandas as pd
+from keras.models import load_model
+from keras.preprocessing import sequence
+
+
+parser = argparse.ArgumentParser()
+parser.add_argument("-l", "--ngrams", type=int, default=3)
+parser.add_argument("-m", "--model_name", default="LSTM")
+args = parser.parse_args()
+
+# Load model
+model = load_model("models/%s.h5" % args.model_name)
+filename = "models/" + args.model_name + "_idx_dic.pkl"
+idx_dic = pickle.load(open(filename, "rb"))
+categories = pd.read_csv("models/%s_categories.txt" % args.model_name, header=None)
+
+# Get PubMed author data from iscb-diversity repo
+df = pd.read_csv("https://github.com/greenelab/iscb-diversity/raw/master/data/pubmed/authors.tsv.xz", sep='\t')
+df.dropna(subset=['fore_name', 'last_name'], inplace=True)
+sdf = df
+sdf['name_last_name_first'] = sdf['last_name'] + ' ' + sdf['fore_name']
+pubmed_names_list = sdf['name_last_name_first']
+
+X_pubmed = utils.featurize_data(pubmed_names_list, args.ngrams, idx_dic)
+X_pubmed = sequence.pad_sequences(X_pubmed, maxlen=20)
+y_pubmed_pred = model.predict_proba(X_pubmed, verbose=2)
+print("PubMed authors",np.mean(y_pubmed_pred, axis=0))
+y_pubmed_prob = pd.DataFrame(y_pubmed_pred, columns=categories[0], index=pubmed_names_list)
+y_pubmed_prob.to_csv("data/%s_results_authors.tsv" % args.model_name, sep='\t')
+
+
+# Get ISMB keynote data from iscb-diversity repo
+df = pd.read_csv("https://raw.githubusercontent.com/greenelab/iscb-diversity/master/data/iscb/keynotes.tsv", sep='\t')
```

Comment:
version here as well.

Looks like we're only making predictions for PubMed and honoree names but not PMC? We will also want to do PMC.